assets/src/hooks/useSocket.ts

```diff
@@ -42,6 +42,10 @@ const useSocket = (): SocketStatus => {
       setConnectionStatus(ConnectionStatus.Disconnected)
     )
     setSocket(initialSocket)
+
+    return () => {
+      setConnectionStatus(ConnectionStatus.Disconnected)
+    }
```

Comment:
I think you're correct that there should be some cleanup here, though I'm not entirely sure exactly what. This cleanup does the same thing as the socket's `onClose` callback, although that may still make a difference since there's nothing to call that `onClose` callback on the mock socket. One other possibility that comes to mind for the cleanup is invoking the [disconnect callback](https://hexdocs.pm/phoenix/js/#socketdisconnect). I also wonder about setting the socket state back to `undefined`, though presumably that doesn't matter much if the hook is being cleaned up due to being unmounted. Going further down the rabbit hole, I wonder if the socket object will simply drop the connection when it's cleaned up in memory management, meaning we just need to make sure it's no longer referenced.

Reply:
I think this just unravelled another layer as to why the original test was producing `Warning: Can't perform a React state update on an unmounted component` - `PropertiesPanel` seems to be the only component that calls `useSocket` directly - all others use the socket from `SocketContext`. 

That indicates to me that in most component tests which don't use the `SocketProvider` (only `App.test.tsx` does), the socket would be undefined, so we'd never encounter this unmounting issue. I'll remove the changes to `useSocket` from this PR & add these details to the ticket -- maybe also worth considering if `PropertiesPanel` should use the socket from context as part of that.

Reply:
Removed the socket changes in [cb2ca68](https://github.com/mbta/skate/pull/1748/commits/cb2ca68c8d7229ec74503cc0f292cbd85657b704) and created a [new ticket](https://app.asana.com/0/1148853526253420/1203118229899945/f) for investigating those considerations